### PR TITLE
Refactor quick menu module

### PR DESCRIPTION
**Pull request by romandev**
This patch refactors quick menu module

This patch refactors overall architecture of the quick menu module to
fix the issues #1004, #1008, #1010 and so on. This change makes the path
of dependencies simpler and clearer. Also, this change removes the heavy
operations such as serialization/deserialization for deepcopy in pending
model update.

**Before:**
```
Client ---> Factory
   |           |
   |------> Controller --------> ViewModel --------> View
               /\  |    update              notify     |
               |   -------------------------------------
               |                                       |
               -----------------------------------------
                               delegate
```

**After:**
```
Client ---> Service ---> View ---> Controller --------> ViewModel
                          /\                   update       |
                          |                                 |
                          -----------------------------------
                                         notify
```

Refs: #976, #1004, #1008, #1010

### CODE REVIEW

**Review by codeimpl — COMMENTED** [review]
LGTM

**Review by codeimpl — APPROVED** [review]
LGTM

File: src/org/triple/banana/quick_menu/View.java [hunk 1]
```diff
@@ -1,11 +1,14 @@
-// Copyright 2020 The Triple Banana Authors. All rights reserved.
+// Copyright 2021 The Triple Banana Authors. All rights reserved.
 // This Source Code Form is subject to the terms of the Mozilla Public
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at https://mozilla.org/MPL/2.0/.
 
 package org.triple.banana.quick_menu;
 
-public interface QuickMenuController {
+import android.content.Context;
```
Comment (jdragonbae) [inline src/org/triple/banana/quick_menu/View.java:8]: Seems like unused import

File: src/org/triple/banana/quick_menu/View.java [hunk 2]
```diff
@@ -1,11 +1,14 @@
...
+
+import androidx.annotation.NonNull;
```
Comment (jdragonbae) [inline src/org/triple/banana/quick_menu/View.java:10]: Seems like unused import

Comment (romandev) [reply src/org/triple/banana/quick_menu/View.java]: Done

**Review by jdragonbae — APPROVED** [review]
LGTM! Simpler architecture :+1:

**Pull request merged**